static/js/server_events_dispatch.js

```diff
@@ -234,6 +235,7 @@ export function dispatch_normal_event(event) {
                         page_params["realm_" + event.property] = event.value;
                         realm_settings[event.property]();
                         settings_org.sync_realm_settings(event.property);
+                        stream_ui_updates.update_notifications_stream_in_settings();
```

Comment:
This should be called only when notification stream setting has changed. You can add this to the `realm_settings` dict just above this `switch` block.